src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+        }
+      }
+      else {
+        console.log("Already started last round or submitted to this round")
+      }
+    }
+
+    //update state
+    await updateTable("jobs", {"last_received_request_id": Number(requestId)}, jobName)
+
+  });
+
+  /**
+   * POST /jobs endpoint
+   * This is used to listen for new jobs added from UI and to update state
+   */
+  app.post('/jobs', async (req, res) => {
+    let newJob = req.body.jobId;
+    let newJobName = req.body.params.name;
+    console.log("new job", newJobName, newJob)
+
+    await createJob(newJob, newJobName);
+
+    res.status(200).send({ success: true })
+  });
+
+  /**
+   * DELETE /jobs/:id endpoint
+   * This is used to listen for jobs deleted from UI and to update state
+   */
+  app.delete('/jobs/:id', async (req, res) => {
+    let jobId = req.params.id;
+    console.log("Removing job", jobId)
+
+    await deleteJob(jobId)
+
+    res.status(200).send({ success: true })
+  });
+ 
+  const listener = app.listen(PORT, '0.0.0.0', () => {
+    console.log(`External adapter listening on port`, PORT);
+  });
+
+  listener.on('error', err => { console.log("Bridge found error:", err) })
+}
+
+/** @param {import('../lib/psm.js').BridgeAction} bridgeAction */
+const outputAction = bridgeAction => {
+  const capData = marshaller.serialize(bridgeAction);
+  var data = JSON.stringify(capData)
+  return data
+};
+
+/**
+ * Function to check if currently in submission
+ * @param {*} feed feed to check for
+ * @returns whether last submission was made in less than SEND_CHECK_INTERVAL seconds
+ */
+const checkIfInSubmission = async (feed) => {
+
+  //get last submission time
+  let query = await queryTable("jobs", ["last_submission_time"], feed)
+  //get seconds since last price submission
+  let timePassedSinceSubmission = (Date.now()/1000) - 
+  query.last_submission_time
+  return timePassedSinceSubmission < Number(SEND_CHECK_INTERVAL)
+}
+
+/**
+  * Function to push price on chain to the smart wallet
+  * @param {*} price price to push
+  * @param {*} feed feed to push price to
+  * @param {*} round round to push result to
+  * @param {*} from account to push from
+  * @returns whether successful
+  */
+const pushPrice = async (price, feed, round, from) => {
+
+  //create an offerId with the Date number
+  var offerId = Date.now()
+
+  //get offers
+  let offers = await getOraclesInvitations()
+  //get previous offer for feed
+  let previousOffer = offers[feed]
+
+  //create an offer
+  const offer = {
+    id: Number(offerId),
+    invitationSpec: {
+      source: 'continuing',
+      previousOffer: Number(previousOffer),
+      invitationMakerName: 'PushPrice',
+      invitationArgs: harden([{ unitPrice: BigInt(price), roundId: round }]),
+    },
+    proposal: {},
+  };
+
+  //create keyring
+  var keyring = {
+    "home": "",
+    "backend": "test"
+  }
+
+  //check if submitted for round
+  let submitted = await checkSubmissionForRound(from, previousOffer, round)
```

Comment:
This was already checked before why calling it again?

Reply:
Before where? This is checked once before submitting the first transaction and after every attempt to avoid sending multiple transactions to avoid extra costs.